src/dispatchers/browserTypeDispatcher.ts

```diff
@@ -47,4 +50,29 @@ export class BrowserTypeDispatcher extends Dispatcher<BrowserType, channels.Brow
       defaultContext: browser._defaultContext ? new BrowserContextDispatcher(browserDispatcher._scope, browser._defaultContext) : undefined,
     };
   }
+
+  async connect(params: channels.BrowserTypeConnectParams): Promise<channels.BrowserTypeConnectResult> {
+    const waitForNextTask = params.slowMo
+      ? (cb: () => any) => setTimeout(cb, params.slowMo)
+      : makeWaitForNextTask();
+    const paramsHeaders = Object.assign({'User-Agent': getUserAgent()}, params.headers || {});
+    const ws = new WebSocket(params.wsEndpoint, [], {
+      perMessageDeflate: false,
+      maxPayload: 256 * 1024 * 1024, // 256Mb,
+      handshakeTimeout: params.timeout || 30000,
+      headers: paramsHeaders,
+    });
+    const pipe = new JsonPipeDispatcher(this._scope);
+    ws.on('open', () => pipe.wasOpened());
+    ws.on('close', () => pipe.wasClosed());
+    ws.on('error', error => pipe.wasClosed(error));
+    pipe.on('close', () => ws.close());
+    pipe.on('message', message => ws.send(JSON.stringify(message)));
+    ws.addEventListener('message', event => {
+      waitForNextTask(() => {
+        pipe.dispatch(JSON.parse(event.data));
```

Comment:
Let's try/catch the JSON.parse and error the pipe.

Reply:
I can close ws on those.